src/ts/InternalBitmovinYospacePlayer.ts

```diff
@@ -455,15 +455,18 @@ export class InternalBitmovinYospacePlayer implements BitmovinYospacePlayerAPI {
     if (this.adImmune) {
       const currentTime = this.player.getCurrentTime();
 
-      this.session?.getAdBreaksByType(BreakType.LINEAR).forEach((adBreak) => {
-        const breakStart = this.toMagicTime(toSeconds(adBreak.getStart()));
+      const linearAdBreaks = this.session?.getAdBreaksByType(BreakType.LINEAR);
+      if (linearAdBreaks !== null) {
+        linearAdBreaks?.forEach((adBreak) => {
```

Comment:
With the changed check I suggested above, you also don't need the `?.` operator here anymore.

```suggestion
        linearAdBreaks.forEach((adBreak) => {
```

Alternatively, we could remove the `if` completely and only keep the `?.` operator here. Doesn't matter too much, whatever you prefer 🤷

Reply:
thank you; removed the if completely and kept the ?. operator..